### PR TITLE
New XML parser on Windows

### PR DESCRIPTION
**Pull request by Jinming-Hu**
# Pull Request Checklist
closes https://github.com/Azure/azure-sdk-for-cpp/issues/2615
closes https://github.com/Azure/azure-sdk-for-cpp/issues/1697

Please leverage this checklist as a reminder to address commonly occurring feedback when submitting a pull request to make sure your PR can be reviewed quickly:

See the detailed list in the [contributing guide](https://github.com/Azure/azure-sdk-for-cpp/blob/master/CONTRIBUTING.md#pull-requests).

- [x] [C++ Guidelines](https://azure.github.io/azure-sdk/cpp_introduction.html)
- [x] Doxygen docs
- [x] Unit tests
- [x] No unwanted commits/changes
- [x] Descriptive title/description
  - [x] PR is single purpose
  - [x] Related issue listed
- [x] Comments in source
- [x] No typos
- [x] Update changelog
- [x] Not work-in-progress
- [x] External references or docs updated
- [x] Self review of PR done
- [x] Any breaking changes?

### CODE REVIEW

**Review by katmsft — CHANGES_REQUESTED** [review]
Add test to cover the newly added path on windows for XML reader/writer.

**Comment by Jinming-Hu** [comment]
> Add test to cover the newly added path on windows for XML reader/writer.

Isn't it already covered by all those APIs that transfer data with XML, Like ListBlobs?

**Review by katmsft — APPROVED** [review]
I don't think the current scenario test covers edge cases for the newly added code. But approve to unblock.

**Review by antkmsft — APPROVED** [review]

File: sdk/storage/azure-storage-common/CHANGELOG.md [hunk 1]
```diff
@@ -4,10 +4,14 @@
 
 ### Features Added
 
+- Used new xml library on Windows, dropped dependency for libxml2.
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/CHANGELOG.md:7]: How is this a new feature?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/CHANGELOG.md:7]: Maybe. But we've shipped the release, it's too late.

File: sdk/storage/azure-storage-common/src/xml_wrapper.cpp [hunk 2]
```diff
@@ -4,13 +4,381 @@
...
+    WS_XML_READER_BUFFER_INPUT bufferInput;
+    ZeroMemory(&bufferInput, sizeof(bufferInput));
+    bufferInput.input.inputType = WS_XML_READER_INPUT_TYPE_BUFFER;
+    bufferInput.encodedData = const_cast<char*>(data);
+    bufferInput.encodedDataSize = static_cast<ULONG>(length);
+    WS_XML_READER_TEXT_ENCODING textEncoding;
+    ZeroMemory(&textEncoding, sizeof(textEncoding));
+    textEncoding.encoding.encodingType = WS_XML_READER_ENCODING_TYPE_TEXT;
+    textEncoding.charSet = WS_CHARSET_AUTO;
+    HRESULT ret = WsSetInput(
```
Comment (ahsonkhan) [inline sdk/storage/azure-storage-common/src/xml_wrapper.cpp:78]: How's the performance? This is something @JeffreyRichter had called out earlier:
https://github.com/Azure/azure-sdk-for-cpp/issues/1697#issuecomment-781651511

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/src/xml_wrapper.cpp:78]: rough measurement shows that it's 3~4 times faster than libxml2.

File: sdk/storage/azure-storage-common/CHANGELOG.md (hunk 1 above)
Comment (ahsonkhan) [reply sdk/storage/azure-storage-common/CHANGELOG.md:7]: No worries, for next time :)

File: sdk/storage/azure-storage-common/src/xml_wrapper.cpp (hunk 2 above)
Comment (ahsonkhan) [reply sdk/storage/azure-storage-common/src/xml_wrapper.cpp:78]: Does it have a noticeable e2e impact on things like blob upload/download?

Comment (Jinming-Hu) [reply sdk/storage/azure-storage-common/src/xml_wrapper.cpp:78]: > Does it have a noticeable e2e impact on things like blob upload/download?

I don't think so. Only list operations return xml body, the size of which is usually several MB. Parsing this size of data can be really fast compared to the time spent on network transfer. So there's no noticeable impact on e2e.
